ibis/expr/operations.py

```diff
@@ -168,6 +175,22 @@ def aggregate(self, this, metrics, by=None, having=None):
     def sort_by(self, expr, sort_exprs):
         return Selection(expr, [], sort_keys=sort_exprs)
 
+    def is_ancestor(self, other):
+        import ibis.expr.lineage as lin
+
+        if isinstance(other, ir.Expr):
+            other = other.op()
+
+        if self.equals(other):
+            return True
+
+        fn = lambda e: (lin.proceed, e.op())  # noqa: E731
```

Comment:
Perhaps It'd be useful to factor out this block to a `lin.flatten` function. I start to like even more this `lin.traverse` :)

Reply:
Cool, can you open an issue for that?

Reply:
Done https://github.com/ibis-project/ibis/issues/1612